### PR TITLE
Fix Issue #9671 

### LINKED ISSUE #9671: Optimizations fails on a specific graph
**Describe the bug**

The optimization fails on the attached ONNX graph. The node mentioned in the error does not belong to the original graph and it works when optimizations are disabled.

`RuntimeError: Unable to create InferenceSession due to '[ONNXRuntimeError] : 9 : NOT_IMPLEMENTED : Could not find an implementation for Transpose(13) node with name 'Transpose_token_1''`

[optimisation_fails.zip](https://github.com/microsoft/onnxruntime/files/7474509/optimisation_fails.zip)

**Urgency**
Before the next release.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- ONNX Runtime installed from (source or binary): orttraining built from github
- ONNX Runtime version: 0.10.1
- Python version: 3.9
- Visual Studio version (if applicable): 2019
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory: none

**To Reproduce**
Create a session with the attached graph.

**Expected behavior**
No failure.

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Fix #9671 by running the level 1 rewrite rules first and allowing the transpose optimizer to run multiple times to ensure it completes in level 1.

Removed unnecessary call to GenerateRuleBasedGraphTransformer as there are no level 2 rewrite rules.

**Motivation and Context**
#9671

### CODE REVIEW

File: onnxruntime/test/optimizer/transpose_optimizer_test.cc [hunk 1]
```diff
@@ -3816,8 +3816,23 @@ TEST(TransposeOptimizerTests, TestOmitIdentityTranspose) {
...
 
+// regression test for a model where the transpose optimizations were not completed in a single pass in level 1.
+// fixed by
+//   a) moving the RewriteRule level 1 optimizations so they run prior to the transpose optimizer; and
+//   b) not returning `true` from TransposeOptimizer::ShouldOnlyApplyOnce as it should be safe to run the
+//      transpose optimizer multiple times to ensure it completes in level 1.
+// either of those changes would have fixed the issue.
+// see https://github.com/microsoft/onnxruntime/issues/9671 for more details.
+TEST(TransposeOptimizerTests, RegressionTest_GitHubIssue9671) {
+  Status status;
```
Comment (edgchen1) [inline onnxruntime/test/optimizer/transpose_optimizer_test.cc:3827]: >Status status; [](http://example.com/codeflow?start=2&length=14)

unused?

File: onnxruntime/core/optimizer/graph_transformer_utils.cc [hunk 2]
```diff
@@ -146,14 +146,21 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(
...
+      session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsDisableQuantQDQ, "0") == "1";
 #ifndef DISABLE_CONTRIB_OPS
-  bool enable_gelu_approximation = session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsEnableGeluApproximation, "0") == "1";
+  bool enable_gelu_approximation =
+      session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsEnableGeluApproximation, "0") == "1";
 #endif
 
   switch (level) {
     case TransformerLevel::Level1: {
+      // RewriteRule transformers are the simplest so run them first
```
Comment (edgchen1) [inline onnxruntime/core/optimizer/graph_transformer_utils.cc:158]: >RewriteRule transformers are the simplest so run them first [](http://example.com/codeflow?start=9&length=59)

perhaps I'm missing something, but why is being simplest a good reason to run them first?

Comment (skottmckay) [reply onnxruntime/core/optimizer/graph_transformer_utils.cc:158]: they tend to remove unnecessary nodes and are lower cost to run, so reduces the work for more complicated optimizers like constant folding (which actually executes nodes so is more expensive).

Comment (skottmckay) [reply onnxruntime/core/optimizer/graph_transformer_utils.cc]: Updated comment to add this detail

**Review by edgchen1 — APPROVED** [review]
:shipit:

**Pull request merged**